File: src/main/java/models/FavorisPost.java
package models;

import java.util.Date;

public class FavorisPost {
    private Favoris favoris;
    private Post post;
    private Date dateAjout;

    public FavorisPost(Favoris favoris, Post post) {
        this.favoris = favoris;
        this.post = post;
        this.dateAjout = new Date();
    }

    public FavorisPost() {

    }

    public Favoris getFavoris() {
        return favoris;
    }

    public void setFavoris(Favoris favoris) {
        this.favoris = favoris;
    }

    public Post getPost() {
        return post;
    }

    public void setPost(Post post) {
        this.post = post;
    }

    public Date getDateAjout() {
        return dateAjout;
    }

    public void setDateAjout(Date dateAjout) {
        this.dateAjout = dateAjout;
    }


}

File: src/Entity/Favoris_post.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

use App\Entity\Post;

#[ORM\Entity]
class Favoris_post
{

    #[ORM\Id]
        #[ORM\ManyToOne(targetEntity: Favoris::class, inversedBy: "favoris_posts")]
    #[ORM\JoinColumn(name: 'favoris_id', referencedColumnName: 'id', onDelete: 'CASCADE')]
    private Favoris $favoris_id;

    #[ORM\Id]
        #[ORM\ManyToOne(targetEntity: Post::class, inversedBy: "favoris_posts")]
    #[ORM\JoinColumn(name: 'post_id', referencedColumnName: 'id', onDelete: 'CASCADE')]
    private Post $post_id;

    #[ORM\Column(name: "dateAjout",type: "datetime")]
    private \DateTimeInterface $dateAjout;

    public function getFavoris_id()
    {
        return $this->favoris_id;
    }

    public function setFavoris_id($value)
    {
        $this->favoris_id = $value;
    }

    public function getPost_id()
    {
        return $this->post_id;
    }

    public function setPost_id($value)
    {
        $this->post_id = $value;
    }

    public function getDateAjout()
    {
        return $this->dateAjout;
    }

    public function setDateAjout($value)
    {
        $this->dateAjout = $value;
    }
}
